swiftnav/src/reference_frame/mod.rs

```diff
@@ -756,29 +953,198 @@ mod tests {
         // Make sure there isn't a direct path
         assert!(!TRANSFORMATIONS.iter().any(|t| t.from == from && t.to == to));
 
-        let graph = TransformationGraph::new();
-        let path = graph.get_shortest_path(from, to);
-        assert!(path.is_some());
+        let graph: TransformationRepository = TransformationRepository::from_builtin();
+        let path = graph.get_shortest_path(&from, &to);
+        assert!(path.is_ok());
         // Make sure that the path is correct. N.B. this may change if more transformations
         // are added in the future
         let path = path.unwrap();
-        assert_eq!(path.len(), 3);
-        assert_eq!(path[0], from);
-        assert_eq!(path[1], ReferenceFrame::ITRF2000);
-        assert_eq!(path[2], to);
+        assert_eq!(path.len(), 2);
     }
 
     #[test]
-    fn fully_traversable_graph() {
-        let graph = TransformationGraph::new();
-        for from in ReferenceFrame::iter() {
-            for to in ReferenceFrame::iter() {
-                if from == to {
-                    continue;
-                }
-                let path = graph.get_shortest_path(from, to);
-                assert!(path.is_some(), "No path from {} to {}", from, to);
-            }
-        }
+    fn transformation_repository_empty() {
+        let repo = TransformationRepository::new();
+        assert_eq!(repo.count(), 0);
+
+        let result = repo.get_shortest_path(&ReferenceFrame::ITRF2020, &ReferenceFrame::ITRF2014);
+        assert!(result.is_err());
+    }
+
+    #[test]
+    fn transformation_repository_from_builtin() {
+        let repo = TransformationRepository::from_builtin();
+        assert_eq!(repo.count(), TRANSFORMATIONS.len() * 2); // Also cound inverted transformations
+
+        // Test path finding
+        let path = repo.get_shortest_path(&ReferenceFrame::ITRF2020, &ReferenceFrame::ETRF2000);
+        assert!(path.is_ok());
```

Comment:
I would encourage not using `assert!(foo.is_ok())` or `assert!(foo.is_err())` in tests if we can help it. The issue is that the assert will swallow up the type. For example, if I assert something is ok, and it's an error, it will not print out the error or any of its information, it will just show that true != false lol

Instead, I would encourage using `unwrap`, `unwrap_err`, or `expect` here instead as it effectively does the same thing (panics if the value is not what we wanted) and will not swallow the error.